### PR TITLE
added react branch

### PR DESCRIPTION
**Pull request by nickav**
Added react-flavored version of this project (might not be 100% correct, I'm pretty new to react)

### CODE REVIEW

File: Gulpfile.js [hunk 1]
```diff
@@ -62,13 +67,24 @@ gulp.task('sass:optimized', function() {
...
 gulp.task('sass', ['sass:lint', 'sass:build']);
 
-gulp.task('js:build', function() {
-  return gulp.src('src/js/**/*.js')
-    .pipe(plumber())
-    .pipe(sourcemaps.init())
-    .pipe(uglify())
-    .pipe(sourcemaps.write())
-    .pipe(gulp.dest('dist/js'));
+gulp.task('webpack:build', function(callback) {
```
Comment (schlosser) [inline Gulpfile.js]: Let's leave this as `js:build`

File: Gulpfile.js [hunk 2]
```diff
@@ -62,13 +67,24 @@ gulp.task('sass:optimized', function() {
...
+  var myConfig = Object.create(webpackConfig);
+  myConfig.plugins = myConfig.plugins.concat(
+    new webpack.DefinePlugin({
+      "process.env": {
+        // This has effect on the react lib size
+        "NODE_ENV": JSON.stringify("production")
+      }
+    }),
+    new webpack.optimize.DedupePlugin(),
+    new webpack.optimize.UglifyJsPlugin()
```
Comment (schlosser) [inline Gulpfile.js]: Is there a more Gulp-y way to do uglifyjs / sourcemaps / etc?  Or do we have to use webpack plugins?

Comment (schlosser) [inline Gulpfile.js]: Moreover, do we need webpack at all? Why not use [`gulp-babel`](https://www.npmjs.com/package/gulp-babel) instead?  I'm not a big react guy but this seems like a lot of extra code just to compile JSX

**Comment by nickaversano** [comment]
I couldn't submit to a non-existing branch, could you create it?

File: Gulpfile.js (hunk 2 above)
Comment (nickaversano) [inline Gulpfile.js]: Not exactly sure myself... I integrated [react-hot-loader](https://github.com/gaearon/react-hot-boilerplate) and that's the minimum requirements for it. There could be a simpler way to do this for sure

Comment (nickaversano) [inline Gulpfile.js]: [this](http://www.bebetterdeveloper.com/coding/es6-react-babel.html) seems like what we're looking for

Comment (schlosser) [inline Gulpfile.js]: [It looks like](https://www.npmjs.com/package/gulp-react) gulp-react is deprecated "in favor of gulp-babel`?

**Comment by schlosser** [comment]
Done: https://github.com/minimill/project-template/tree/react

File: Gulpfile.js (hunk 2 above)
Comment (nickaversano) [inline Gulpfile.js]: OK, so [this](https://jonsuh.com/blog/integrating-react-with-gulp/) looks like a more minimal setup but I couldn't figure out how to get it to work

**Pull request closed without merging**